src/univariate/continuous/triangular.jl

```diff
@@ -93,20 +93,34 @@ entropy(d::TriangularDist{T}) where {T<:Real} = one(T)/2 + log((d.b - d.a) / 2)
 
 function pdf(d::TriangularDist, x::Real)
     a, b, c = params(d)
-    return if x < c
-        2 * max(x - a, 0) / ((b - a) * (c - a))
+    T = typeof(one(x) / one(a))
+    if x < a || b < x
+        return T(0)
+    elseif a <= x < c
+        return 2 * (x - a) / ((b - a) * (c - a))
+    elseif x == c
+        return T(2) / (b - a)
+    elseif c < x <= b
+        return 2 * (b - x) / ((b - a) * (b - c))
     else
-        2 * max(b - x, 0) / ((b - a) * (b - c))
+        return T(NaN)
```

Comment:
Can you remove this branch? I don't think https://github.com/JuliaStats/Distributions.jl/pull/1674/files#r1101068580 is resolved - IMO the tests should not be fixed by adding a `NaN` branch but by changing the last `elseif` to an `else`.